File: src/main/java/com/cuupa/classificator/services/kb/semantic/dataExtraction/SenderExtract.java
package com.cuupa.classificator.services.kb.semantic.dataExtraction;

import org.jetbrains.annotations.NotNull;

import java.util.regex.Pattern;

/**
 * @author Simon Thiel (https://github.com/cuupa)
 */
public class SenderExtract extends Extract {

    private static Pattern pattern;

    public SenderExtract(@NotNull String regex) {
        pattern = Pattern.compile(regex);
    }

    @Override
    public Pattern getPattern() {
        return pattern;
    }

    @NotNull
    @Override
    public String normalize(@NotNull String value) {
        return value.trim();
    }
}

File: src/main/java/com/cuupa/classificator/services/kb/semantic/dataExtraction/SenderExtract.kt
package com.cuupa.classificator.services.kb.semantic.dataExtraction

import java.util.regex.Pattern

/**
 * @author Simon Thiel (https://github.com/cuupa)
 */
class SenderExtract(regex: String) : Extract(Pattern.compile(regex.trim(), Pattern.CASE_INSENSITIVE)) {

    override fun normalize(value: String): String {
        return value.trim()
    }

    companion object {
        const val name = "[SENDER]"
    }
}